Initial create

charlotta-frends opened the pull request:
- Tested in Frends with frameworks standard, net6 & net8 on Windows-based agents
- Versioning starts from 1.1.0 because some tenant already has v1.0.x

charlotta-frends committed:
initial create

charlotta-frends committed:
Dependency injection for converter, added frameworks, cleaning

charlotta-frends committed:
fix workflows

charlotta-frends committed:
Adjustments to error handling, more cleanup

charlotta-frends committed:
Fix workflows

jefim commented on .github/workflows/BuildAndTestOnEveryPush.yml:
```diff
@@ -0,0 +1,28 @@
+    
+    - name: Test
+      env:
+        EXAMPLE_ENVIROMENT_VARIABLE:  ${{ secrets.EXAMPLE_ENVIROMENT_VARIABLE }}
```
This is probably not needed

jefim commented on Frends.Community.HTMLToPdf/Definitions/Input.cs:
```diff
@@ -0,0 +1,28 @@
+    /// <example>C:\temp\foo.pdf"</example>
+    [DisplayFormat(DataFormatString = "Text")]
+    [DefaultValue(@"C:\temp\foo.pdf")]
+    public string OutputPath { get; set; }
```
This would be better to be an option like `OutputMode = [File/Bytes]`, depending on which the OutputPath could be shown/hidden. Or a checkbox, whichever feels better.

jefim commented on Frends.Community.HTMLToPdf/Frends.Community.HTMLToPdf.cs:
```diff
@@ -0,0 +1,90 @@
+
+            return new Result(true, input.OutputPath, null);
+        }
+        catch (Exception ex)
```
What does this exception handler do? Either remove or add meaningful context to the error. Also, if you add context for it wrap the original exception as an inner exception.

jefim commented on README.md:
```diff
@@ -1 +1,86 @@
+
+| Version | Changes |
+| ------- | ------- |
+| 1.1.0   | Initial create |
```
Why is initial version immediately 1.1.0?

jefim commented on Frends.Community.HTMLToPdf/Frends.Community.HTMLToPdf.csproj:
```diff
@@ -0,0 +1,39 @@
+  <PropertyGroup>
+	  <TargetFrameworks>netstandard2.0;net6.0;net8.0</TargetFrameworks>
+	  <LangVersion>Latest</LangVersion>
+	  <Version>1.1.0</Version>
```
Not 1.0.0?

jefim commented on Frends.Community.HTMLToPdf/Frends.Community.HTMLToPdf.csproj:
```diff
@@ -0,0 +1,39 @@
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Haukcode.WkHtmlToPdfDotNet" Version="1.5.93" />
```
This library uses LGPL 3.0 apparently, please update the task to comply with LGPL 3.0 requirements.

charlotta-frends committed:
Review changes

- Removed unnecessary env var from workflows
- Added OutputType for conditional visibility of the OutputPath & updated readme
- Added context and wrapping for the conversion exception
- Added mention of WkHtmlToPdf-DotNet's LGPL-3.0 licence & how to comply with it, and added LGPL-3.0 licence text as a separate file

charlotta-frends commented:
Review changes:
- Removed unnecessary env var from workflows
- Added OutputType for conditional visibility of the OutputPath & updated readme
- Added context and wrapping for the conversion exception
- Added mention & instructions of WkHtmlToPdf-DotNet's LGPL-3.0 licence, and added LGPL-3.0 licence text as a separate file. Since the library is dynamically linked here, this should comply with the license requirements.
